File: week1/Can-Yavuz(a.k.a=TuvaletPompasi)/twoSumII.ts
function twoSum(numbers: number[], target: number): number[] {
    let left = 0;
    let right = numbers.length - 1;
    while (left < right) {
        const sum = numbers[left] + numbers[right];
        if (sum === target) {
            return [left + 1, right + 1]; 
            left++;
        } else {
            right--;
        }
    }
    return [];
}

File: week1/Bu$ra/TwoSumII.cs
public class Solution {
    public int[] TwoSum(int[] numbers, int target) {
        int left = 0;
        int right = numbers.Length - 1;

        while (left < right) {
            int sum = numbers[left] + numbers[right];

            if (sum == target) {
                return new int[] { left + 1, right + 1 };
            } else if (sum < target) {
                left++;
            } else {
                right--;
            }
        }

        // This line is here to satisfy the compiler, as there is always exactly one solution.
        return new int[] { -1, -1 };
    }
}
    
